From 28b42ef49bd5352a1cbf62517fe80fdb8f1f2ced Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 22 Apr 2022 10:01:55 +0000
Subject: [PATCH] Fix build with gcc-10

:Release Notes:
Fixes:
http://gecko.lge.com/Errors/Details/384977

:Detailed Notes:
qemux86_64-webos-linux/com.webos.service.memorymanager/1.0.0-52-r6/git/src/common/util/Cgroup.cpp: In static member function 'static void Cgroup::iterateDir(std::map<std::__cxx11::basic_string<char>, std::__cxx11::list<int> >&, std::string)':
qemux86_64-webos-linux/com.webos.service.memorymanager/1.0.0-52-r6/git/src/common/util/Cgroup.cpp:49:18: error: variable 'std::ifstream ifs' has initializer but incomplete type
   49 |     ifstream ifs(procs.c_str());
      |                  ^~~~~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-10791] CCC: Various build fixes

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/com.webos.service.memorymanager/+/347408 Fix build with gcc-10]

Change-Id: Ia254ca5a9bb40b345e148acb49a611a9a43b1b33
---
 src/common/util/Cgroup.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/common/util/Cgroup.cpp b/src/common/util/Cgroup.cpp
index ad4f4e0..478c4ab 100644
--- a/src/common/util/Cgroup.cpp
+++ b/src/common/util/Cgroup.cpp
@@ -15,6 +15,8 @@
 
 #include "Cgroup.h"
 
+#include <fstream>
+
 #include <boost/filesystem.hpp>
 #include <boost/algorithm/string.hpp>
 
